dvc/repo/params/show.py

```diff
@@ -117,74 +95,101 @@ def _collect_vars(repo, params, stages=None) -> Dict:
     return vars_params
 
 
+def _read_params(
+    fs: "FileSystem", params: Dict[str, List[str]]
+) -> Iterator[Tuple[str, Union[Exception, Any]]]:
+    for file_path, key_paths in params.items():
+        try:
+            yield file_path, read_param_file(fs, file_path, key_paths)
+        except Exception as exc:  # noqa: BLE001 # pylint:disable=broad-exception-caught
+            logger.debug(exc)
+            yield file_path, exc
+
+
+def _gather_params(
+    repo: "Repo",
+    targets: Union[List[str], Dict[str, List[str]], None] = None,
+    deps_only: bool = False,
+    stages: Optional[List[str]] = None,
+    on_error: str = "return",
+):
+    assert on_error in ("raise", "return", "ignore")
+
+    files_keypaths = _collect_params(
+        repo,
+        targets=targets,
+        stages=stages,
+        deps_only=deps_only,
+        default_file=ParamsDependency.DEFAULT_PARAMS_FILE,
+    )
+
+    data: Dict[str, FileResult] = {}
+
+    dvcfs = repo.dvcfs
+    for fs_path, result in _read_params(dvcfs, files_keypaths):
+        repo_path = fs_path.lstrip(dvcfs.root_marker)
+        repo_os_path = os.sep.join(repo.fs.path.parts(repo_path))
+        if not isinstance(result, Exception):
+            data.update({repo_os_path: FileResult(data=result)})
+            continue
+
+        if on_error == "raise":
+            raise result
+        if on_error == "return":
+            data.update({repo_os_path: FileResult(error=result)})
+
+    # vars_params = _collect_vars(repo, params, stages=stages)
```

Comment:
We used to show `params` from the variables used in the templating (other than from `params.yaml` file). But determining the true source has been challenging (especially when the paths have to be consistent), and merging all of them is challenging. The implementation is incomplete too (it is skipping any file that is already read, but params are granular that it's not enough).

We have top-level params now, that this does not feel necessary or useful.

With templating variable, i meant this:
```yaml
# params1.yaml
foo: foobar

# dvc.yaml
vars:
- params1.yaml

stages:
  stage1:
    cmd: echo ${foobar}
```

```console
$ dvc params diff
Path          Param    HEAD    workspace                              
params1.yaml  foo      -       foobar
```

But the implementation is incomplete, this was only shown when the file is not specified anywhere as params.

Reply:
🤔 This seems like it will be surprising for people who are using features like dict unpacking. Would they now need to explicitly list all of the unpacked args in params?

Reply:
Brought that back.